src/test/java/database/service/service/JsonDatabaseServiceTest.java

```diff
@@ -0,0 +1,290 @@
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+
+        NullPropertyNameOrValueException exception = assertThrows(NullPropertyNameOrValueException.class, () ->
+                jsonDatabaseService.getByFilters(Student.class, filters));
+        assertEquals("Property name and value cannot be null.", exception.getMessage());
+
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void getByFiltersValueIsNullTest() {
+        Set<Pair<String, Object>> filters = new HashSet<>();
+        filters.add(new ImmutablePair<>("averageScore", null));
+
+        jsonDatabaseService.createTable(Student.class);
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+
+        NullPropertyNameOrValueException exception = assertThrows(NullPropertyNameOrValueException.class, () ->
+                jsonDatabaseService.getByFilters(Student.class, filters));
+        assertEquals("Property name and value cannot be null.", exception.getMessage());
+
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void getByFiltersValueIsEmptyTest() {
+        Set<Pair<String, Object>> filters = new HashSet<>();
+        filters.add(new ImmutablePair<>("fullName", ""));
+
+        jsonDatabaseService.createTable(Student.class);
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+
+        EmptyValueException exception = assertThrows(EmptyValueException.class, () ->
+                jsonDatabaseService.getByFilters(Student.class, filters));
+        assertEquals("Value cannot be empty.", exception.getMessage());
+
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void getByFiltersIncorrectPropertyNameTest() {
+        Set<Pair<String, Object>> filters = new HashSet<>();
+        filters.add(new ImmutablePair<>("firstName", "Mikhail"));
+
+        jsonDatabaseService.createTable(Student.class);
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+
+        IncorrectPropertyNameException exception = assertThrows(IncorrectPropertyNameException.class, () ->
+                jsonDatabaseService.getByFilters(Student.class, filters));
+        assertEquals("Incorrect property name: firstName", exception.getMessage());
+
+        assertTrue(jsonDatabaseService.deleteTable(Student.class));
+    }
+
+    @Test
+    public void getByFiltersIncorrectValueTypeTest() {
+        Set<Pair<String, Object>> filters = new HashSet<>();
+        filters.add(new ImmutablePair<>("fullName", 100));
+
+        jsonDatabaseService.createTable(Student.class);
+        jsonDatabaseService.addNewRecordToTable(firstStudent);
+
+        IncorrectValueTypeException exception = assertThrows(IncorrectValueTypeException.class, () ->
+                jsonDatabaseService.getByFilters(Student.class, filters));
+        assertEquals("Incorrect value type for filter: fullName. Expected: java.lang.String, " +
```

Comment:
You can simplify this bu using `assertTrue(exception.getMessage().contains("Incorrect value type for filter: fullName"));`